pkg/internal/controller/controller.go

```diff
@@ -303,7 +305,12 @@ func (c *Controller) reconcileHandler(ctx context.Context, obj interface{}) {
 		return
 	}
 
-	log := c.Log.WithValues("name", req.Name, "namespace", req.Namespace)
+	var log logr.Logger
+	if c.LoggerCustomizer != nil {
+		log = c.LoggerCustomizer(c.Log, req)
+	} else {
+		log = c.Log.WithValues("name", req.Name, "namespace", req.Namespace)
+	}
 	ctx = logf.IntoContext(ctx, log)
```

Comment:
```suggestion
	var log logr.Logger
	if c.LoggerCustomizer != nil {
		ctx, log = c.LoggerCustomizer(ctx, c.Log, req)
	} else {
		log = c.Log.WithValues("name", req.Name, "namespace", req.Namespace)
		ctx = logf.IntoContext(ctx, log)
	}
```

I wonder if we should make this a bit more powerful. This way the `LoggerCustomizer` could 
use additional context to customize the ctx and the logger (obviously the name doesn't fit that good anymore).

I'm especially thinking about the tracing use case. By passing the ctx in and out of the ctx it would be possible to:
* use the traceID which is potentially passed in with the context in the future
* create a new trace / span and add it to the context

(P.S. this could almost also be done inside of the Reconcile func, but then the log.Error wouldn't have the traceID as k/v pair)